pulp_smash/tests/platform/api_v2/test_repository.py

```diff
@@ -3,25 +3,114 @@
+        │   ├─── it is possible to read an individual distributor of a repo
+        │   │      (via distributor call)
+        │   ├─── It is possible to update a distributor on a repo.
+        │   │    └─── it is possible to read the updated distributor
+        │   └─── It is possible to delete a distributor from a repo.
+        └── It is possible to add an importer to an untyped repo
+            ├─── it is possible to read importers of a repo (via repo call)
+            ├─── it is possible to read importers of a repo
+            │      (via importers call)
+            ├─── it is possible to read an individual importer of a repo
+            │      (via importer call)
+            ├─── It is possible to update a distributor on a repo.
+            │    └─── it is possible to read the updated distributor
+            └─── It is possible to delete a distributor from a repo.
 
 .. _repository:
     https://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/index.html
 
 """
 from __future__ import unicode_literals
+import copy
 
 import requests
 
 from pulp_smash.config import get_config
 from pulp_smash.constants import REPOSITORY_PATH, ERROR_KEYS
 from pulp_smash.utils import uuid4
-from unittest2 import TestCase
+from unittest2 import TestCase, skip
+
+from sys import version_info
+if version_info.major == 2:
+    from urllib import urlencode  # pylint:disable=no-name-in-module
+else:
+    from urllib.parse import urlencode  # noqa pylint:disable=no-name-in-module,import-error
+
+# pylint:disable=duplicate-code
+# Once https://github.com/PulpQE/pulp-smash/pull/28#discussion_r44172668
+# is resolved, pylint can be re-enabled.
+
+ISO_DISTRIBUTOR = {
+    'distributor_id': 'iso_distributor',
+    'distributor_type_id': 'iso_distributor',
+    'distributor_config': {},
+    'auto_publish': True
+}
+
+SERIALIZED_ISO_DISTRIBUTOR = {
+    '_href': '/pulp/api/v2/repositories/{repo}/distributors/iso_distributor/',
+    'auto_publish': True,
+    'config': {},
+    'id': 'iso_distributor',
+    'distributor_type_id': 'iso_distributor',
+    'last_publish': None,
+}
+
+ISO_IMPORTER_TYPE_ID = 'iso_importer'
+
+SERIALIZED_ISO_IMPORTER = {
+    'id': 'iso_importer',
+    'importer_type_id': 'iso_importer',
+    '_href': '/pulp/api/v2/repositories/{repo}/importers/iso_importer/'
+}
+
+
+class RepoBaseTest(TestCase):
+
+    @classmethod
+    def get(cls, path, query=None):
+        """Build a url and make a get request."""
+        if isinstance(query, dict):
+            query = urlencode(query)
+        if query is not None:
+            full_url = "{base}{path}?{query}".format(base=cls.cfg.base_url,
+                                                     path=path, query=query)
+        else:
+            full_url = "{base}{path}".format(base=cls.cfg.base_url, path=path)
+        return requests.get(full_url, **cls.cfg.get_requests_kwargs())
+
+    def _build_expected(self, expected_template, repo_id):
```

Comment:
This is an interesting case. This instance method doesn't access `self`, which indicates that it should either be a `@staticmethod` or a function.